### PR TITLE
Logging config structs

### PR DESCRIPTION
**Pull request by Caleb-Hill**
In an effort to split up pr #1319 for easier review I have pulled out the logging configuration structures, and feature declarations into this pull request.

### CODE REVIEW

File: splinterd/src/logging/mod.rs [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+#[derive(Clone, Debug)]
+pub enum LogTarget {
+    Stdout,
+    Stderr,
+    File(String),
+    RollingFile(String),
+}
+
+fn default_pattern() -> String {
```
Comment (vaporos) [inline splinterd/src/logging/mod.rs:43]: This function could be omitted.

Comment (Caleb-Hill) [reply splinterd/src/logging/mod.rs]: Right you are

**Review by vaporos — DISMISSED** [review]

File: splinterd/src/logging/mod.rs [hunk 2]
```diff
@@ -0,0 +1,40 @@
+use log::Level;
```
Comment (peterschwarz) [inline splinterd/src/logging/mod.rs:1]: This file is missing the copyright header

**Review by peterschwarz — CHANGES_REQUESTED** [review]

File: splinterd/src/logging/mod.rs (hunk 2 above)
Comment (Caleb-Hill) [reply splinterd/src/logging/mod.rs:15]: Fixed

**Review by peterschwarz — APPROVED** [review]

**Review by agunde406 — APPROVED** [review]

**Pull request merged**